src/main/java/it/pagopa/transactions/client/PaymentGatewayClient.java

```diff
@@ -131,6 +133,67 @@ public Mono<XPayAuthResponseEntityDto> requestXPayAuthorization(AuthorizationReq
+                                        .mapToInt(paymentNotice -> paymentNotice.transactionAmount().value()).sum())
+                                        + authorizationData.fee()
+                        );
+                        creditCardAuthRequest = Mono.just(
+                                new CreditCardAuthRequestDto()
+                                        .pan(cardData.getPan())
+                                        .expireDate(cardData.getExpiryDate())
+                                        .idTransaction(
+                                                authorizationData.transaction().getTransactionId().value().toString()
+                                        )
+                                        .amount(grandTotal)
+                                        .emailCh(authorizationData.transaction().getEmail().value())
+                                        .holder(cardData.getHolderName())
+                                        .securitycode(cardData.getCvv())
+                                        .isFirstPayment(true) // TODO TO BE CHECKED
+                                        .threeDsData("threeDsData") // TODO From client
+                                        .circuit("VISA") // TODO Add field to card details
+                                        .reqRefNumber("reqRefNumber") // TODO From client
+                                        .idPsp(authorizationData.pspId())
+                        );
+                    } else {
+                        creditCardAuthRequest = Mono.error(
+                                new InvalidRequestException(
+                                        "Cannot perform VPOS authorization for null input CreditCardAuthRequestDto"
+                                )
+                        );
+                    }
+                    return creditCardAuthRequest;
+                })
+                .flatMap(
+                        creditCardAuthRequestDto -> creditCardInternalApiClient
+                                .step0CreditCard(
+                                        "Web",
+                                        creditCardAuthRequestDto,
+                                        encodeMdcFields(authorizationData)
+                                )
+                                .onErrorMap(
+                                        WebClientResponseException.class,
+                                        exception -> switch (exception.getStatusCode()) {
+                                        case UNAUTHORIZED -> new AlreadyProcessedException(
+                                                authorizationData.transaction().getTransactionId()
+                                        ); // 401
+                                        case INTERNAL_SERVER_ERROR -> new BadGatewayException(""); // 500
+                                        default -> exception;
```

Comment:
502?

Reply:
This are the listed error in the documentation

     * 400 - Bad request - missing mandatory parameters
     * 401 - transactionId already processed
     * 500 - generic error